src/components/Sort/index.js

```diff
@@ -0,0 +1,88 @@
+import { FormControl, Typography, Button } from "@material-ui/core";
+import Image from "next/image";
+import PropTypes from "prop-types";
+import React from "react";
+
+import useStyles from "./useStyles";
+
+import SortDown from "@/twoopstracker/assets/icons/sort-down.svg";
+import SortUp from "@/twoopstracker/assets/icons/sort-up.svg";
+import TweetSelect from "@/twoopstracker/components/TweetSelect";
+
+function Sort({
+  handleSelection,
+  toggleSortOrder,
+  isDesc,
+  label,
+  name,
+  menuItems,
+  value,
+  ...props
+}) {
+  const classes = useStyles(props);
+
+  const handleChange = (event) => {
+    handleSelection({
+      name: name || label.toLowerCase(),
+      value: event.target.value,
+    });
+  };
+
+  return (
+    <FormControl fullWidth className={classes.form}>
+      <Typography variant="body1" className={classes.label}>
+        {label}
+      </Typography>
+      <TweetSelect
+        name={name}
+        label={label}
+        value={value}
+        handleChange={handleChange}
+        menuItems={menuItems}
+      />
+      <Button
+        color="default"
+        variant="text"
+        size="large"
+        onClick={toggleSortOrder}
+        classes={{
+          root: classes.buttonIcon,
+        }}
+        startIcon={
+          <Image
+            layout="fill"
+            src={isDesc ? SortDown : SortUp}
+            className={classes.large}
+          />
+        }
+      />
+    </FormControl>
+  );
+}
+
+Sort.propTypes = {
+  toggleSortOrder: PropTypes.func,
+  isDesc: PropTypes.bool,
+  handleSelection: PropTypes.func,
```

Comment:
Use `onSomething` to show that the component will fire a given event and the necessary party should pass the handle. Use `handleSomething` when you're actually handling the event.

```suggestion
  onClickSortOrder: PropTypes.func,
  isDesc: PropTypes.bool,
  onChangeSortField: PropTypes.func,
```

Or something like that.